packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts

```diff
@@ -183,10 +191,85 @@ export default class CKBoxImageEditCommand extends Command {
 		} );
 	}
 
-	private _waitForAssetProcessed( asset: CKBoxRawAssetDefinition ): Promise<void> {
-		// Timeout for demo purposes.
-		return new Promise( resolve => {
-			setTimeout( resolve, 3000 );
+	/**
+	 * Get asset's status on server. If server respond with "success" status then
+	 * image is already proceeded and ready for saving.
+	 *
+	 * @param data Data about certain asset.
+	 */
+	public _getAssetStatusFromServer( data: CKBoxRawAssetDataDefinition ):
+		Promise<{
+			status: string;
+		}>
+	{
+		const url = new URL( 'assets/' + data.id, this.editor.config.get( 'ckbox.serviceOrigin' )! );
+		const formData = new FormData();
+		const requestConfig = {
+			url,
+			data: formData
+		} as const;
+
+		return this._sendHttpRequest( requestConfig ).then( async res => {
+			return {
+				status: res.metadata.metadataProcessingStatus
+			};
+		} ).catch( err => {
+			return err && Promise.reject( err.message );
+		} );
+	}
+
+	/**
+	 * Waiting until asset is being processed.
+	 *
+	 * @param asset Data about certain asset.
+	 */
+	private _waitForAssetProcessed( asset: CKBoxRawAssetDefinition ): Promise<{ status: string }> {
+		return retry( () => new Promise( ( resolve, reject ) => {
+			this._getAssetStatusFromServer( asset.data ).then( ( data: {
+				status: string;
+			} ) => {
+				if ( data.status === 'success' ) {
+					resolve( data );
+				} else {
+					reject();
+				}
+			} ).catch( err => reject( err ) );
+		} ) );
+	}
+
+	/**
+	 * Sends the HTTP request.
+	 *
+	 * @param config.url the URL where the request will be sent.
+	 * @param config.method The HTTP method.
+	 * @param config.data Additional data to send.
+	 */
+	private _sendHttpRequest( { url, method = 'GET', data }: {
+		url: URL;
+		method?: 'GET' | 'POST';
+		data?: FormData | null;
+	} ) {
+		const ckboxEditing = this.editor.plugins.get( CKBoxEditing );
+		const xhr = new XMLHttpRequest();
+
+		xhr.open( method, url.toString(), true );
```

Comment:
XHR is async by default.

```suggestion
		xhr.open( method, url.toString() );
```